models/DataObject/ClassDefinition/Data/Link.php

```diff
@@ -30,6 +30,21 @@ class Link extends Data implements ResourcePersistenceAwareInterface, QueryResou
     use DataObject\Traits\SimpleComparisonTrait;
     use DataObject\Traits\ObjectVarTrait;
 
+    /**
+     * @var string[]
+     */
+    public array $allowedTypes;
```

Comment:
Shouldn't the default be null if not set? So that everything is then allowed?
So it will throw an error for existing definitions or definition via php script.

Reply:
mhh nice point, didn't consider that.. Can it be just a an empty array by default or does it have drawbacks compared to being null and nullable?